src/utility/misc.h

```diff
@@ -99,9 +99,9 @@ namespace Utility
 	}
 
 	template<typename T>
-	inline int getFirstValueInContainer(const boost::unordered_set<T> &container)
+	inline int getFirstValueInContainer(const std::unordered_set<T> &container)
 	{
-		boost::unordered_set<int>::iterator i = container.begin();
+		auto i = container.begin();
 		if (i != container.end())
```

Comment:
Better. I'd definitely recommend using `auto` (particularly `const auto&`) in more places and wherever possible, especially with iterators.

You can also just initialize `i` directly in the if statement in C++17:
```suggestion
		if (const auto &i = container.begin(); i != container.end())
```
I'm sure there are tons of places where you could do this, so only fix these if you want to.